x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/authn/SuccessfulAuthenticationResponseMessageBuilder.java

```diff
@@ -199,6 +230,16 @@ private AttributeStatement buildAttributes(UserServiceAuthentication user) {
         if (name != null) {
             attributes.add(name);
         }
+        // Add custom attributes if provided
+        if (customAttributes != null && customAttributes.getAttributes().isEmpty() == false) {
+            for (Map.Entry<String, List<String>> entry : customAttributes.getAttributes().entrySet()) {
+                Attribute attribute = buildAttribute(entry.getKey(), entry.getKey(), entry.getValue());
```

Comment:
I think we should skip the `friendlyName` rather than set it to the same value as the formal name.